components/camel-dynamic-router/src/main/java/org/apache/camel/component/dynamicrouter/DynamicRouterMulticastProcessor.java

```diff
@@ -0,0 +1,292 @@
+     */
+    private static final String LOG_ENDPOINT = "log:%s.%s?level=%s&showAll=true&multiline=true";
+
+    private boolean ignoreInvalidEndpoints;
+
+    private final ProducerCache producerCache;
+
+    /**
+     * {@link FilterProcessor}s, mapped by subscription ID, to determine if the incoming exchange should be routed based
+     * on the content.
+     */
+    private final TreeMap<String, PrioritizedFilter> filterMap;
+
+    /**
+     * Indicates the behavior of the Dynamic Router when routing participants are selected to receive an incoming
+     * exchange. If the mode is "firstMatch", then the exchange is routed only to the first participant that has a
+     * matching predicate. If the mode is "allMatch", then the exchange is routed to all participants that have a
+     * matching predicate.
+     */
+    private final String recipientMode;
+
+    /**
+     * The {@link FilterProcessor} factory.
+     */
+    private final Supplier<PrioritizedFilter.PrioritizedFilterFactory> filterProcessorFactorySupplier;
+
+    /**
+     * Flag to log a warning if a message is dropped due to no matching filters.
+     */
+    private final boolean warnDroppedMessage;
+
+    public DynamicRouterMulticastProcessor(String id, CamelContext camelContext, Route route, String recipientMode,
+                                           final boolean warnDroppedMessage,
+                                           final Supplier<PrioritizedFilter.PrioritizedFilterFactory> filterProcessorFactorySupplier,
+                                           ProducerCache producerCache, AggregationStrategy aggregationStrategy,
+                                           boolean parallelProcessing, ExecutorService executorService,
+                                           boolean shutdownExecutorService,
+                                           boolean streaming, boolean stopOnException,
+                                           long timeout, Processor onPrepare, boolean shareUnitOfWork,
+                                           boolean parallelAggregate) {
+        super(camelContext, route, new ArrayList<>(), aggregationStrategy, parallelProcessing, executorService,
+              shutdownExecutorService,
+              streaming, stopOnException, timeout, onPrepare,
+              shareUnitOfWork, parallelAggregate);
+        setId(id);
+        this.producerCache = producerCache;
+        this.filterMap = new TreeMap<>();
+        this.recipientMode = recipientMode;
+        this.filterProcessorFactorySupplier = filterProcessorFactorySupplier;
+        this.warnDroppedMessage = warnDroppedMessage;
+    }
+
+    public boolean isIgnoreInvalidEndpoints() {
+        return ignoreInvalidEndpoints;
+    }
+
+    public void setIgnoreInvalidEndpoints(boolean ignoreInvalidEndpoints) {
+        this.ignoreInvalidEndpoints = ignoreInvalidEndpoints;
+    }
+
+    protected List<Processor> createEndpointProcessors(Exchange exchange) {
```

Comment:
This method is very likely too large to be inlined by the JIT compiler. I haven't analyzed, but I suspect it could be on the hot path ... so, ideally, it should be broken in smaller pieces.

Reply:
Thanks.  I simplified some of this, so please let me know if it looks better.